src/machines/modelingMachine.ts

```diff
@@ -3780,19 +3780,21 @@ export const modelingMachine = setup({
           return Promise.reject(new Error(NO_INPUT_PROVIDED_MESSAGE))
         }
 
-        // Remove once it isn't experimental anymore
+        // Remove once this command isn't experimental anymore
+        let astWithNewSetting: Node<Program> | undefined
         if (kclManager.fileSettings.experimentalFeatures?.type !== 'Allow') {
-          const result = await setExperimentalFeatures({ type: 'Allow' })
-          if (err(result)) {
-            return Promise.reject(result)
+          const ast = setExperimentalFeatures({ type: 'Allow' })
+          if (err(ast)) {
+            return Promise.reject(ast)
           }
+
+          astWithNewSetting = ast
```

Comment:
This was the problematic part in Hole. Now, since we have an ast, we don't even have to run two executions here, but rather pass it directly to the codemod, see below